Card/src/main/java/com/paypal/android/card/network/CardAPIClient.kt

```diff
@@ -0,0 +1,42 @@
+package com.paypal.android.card.network
+
+import com.paypal.android.card.Card
+import com.paypal.android.card.CardError
+import com.paypal.android.card.CardResponse
+import com.paypal.android.card.CardResult
+import com.paypal.android.core.APIClient
+import org.json.JSONObject
+import java.net.HttpURLConnection.HTTP_OK
+
+class CardAPIClient(private val api: APIClient) {
+
+    suspend fun confirmPaymentSource(orderId: String, card: Card): CardResult {
+        val path = "v2/checkout/orders/$orderId/confirm-payment-source"
+        val body = """
+            {
+                "payment_source": {
+                    "card": {
+                        "number": "${card.number}",
+                        "expiry": "${card.expiry}"
+                    }
+                }
+            }
+        """.trimIndent()
+
+        val httpResponse = api.post(path, body)
+        return if (httpResponse.status == HTTP_OK) {
+            val bodyJson = httpResponse.body?.let { JSONObject(it) } ?: JSONObject()
+            val status = bodyJson.getString("status")
```

Comment:
Should these all be `optString` instead? I think we'd get errors if `bodyJson` is instantiated as an empty JSONObject object above

Reply:
Yeah that's true as well. Kotlin with the unchecked exceptions makes it tricky to spot those errors. We talked about the possibility of separating this out into a parser too. I guess we would need to return an error when the response is malformed.